sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextFactory.java

```diff
@@ -243,6 +247,23 @@ private static boolean isScopedToSinglePartition(CosmosQueryRequestOptions cosmo
                    && cosmosQueryRequestOptions.getPartitionKey() != PartitionKey.NONE;
     }
 
+    private static List<FeedRangeEpkImpl> resolveFeedRangeBasedOnPrefixContainer(
+        List<FeedRangeEpkImpl> feedRanges,
+        PartitionKeyDefinition partitionKeyDefinition,
+        PartitionKey partitionKey) {
+        List<FeedRangeEpkImpl> feedRanges2 = new ArrayList<>();
+        PartitionKeyInternal partitionKeyInternal = ModelBridgeInternal.getPartitionKeyInternal(partitionKey);
+        for (FeedRangeEpkImpl feedRange : feedRanges) {
+            if (partitionKeyInternal.getComponents().size() >= partitionKeyDefinition.getPaths().size()) {
+                feedRanges2.add(feedRange);
```

Comment:
I don't get this one? Shouldn't we in that case just return the EPK for the single logical partition of the full pkInternal?